### ISSUE TITLE
Rule suggestion: enforce parentheses for multiple line expressions

### ISSUE THREAD

**Issue by OliverJAsh** [palantir/tslint#2905]
When expressions span multiple lines, sometimes it is desirable to wrap them in parentheses for readability, for example, instead of:

``` ts
export const jsonDecodeString: jsonDecodeString = type => value =>
    // Widen Left generic
    typecheck<Either<JsonDecodeError, any>>(jsonParse(value)).chain(validateType(type));
```

You would wrap the function body in parentheses:

``` ts
export const jsonDecodeString: jsonDecodeString = type => value => (
    // Widen Left generic
    typecheck<Either<JsonDecodeError, any>>(jsonParse(value)).chain(validateType(type))
);
```

Would it be possible to add a rule to enforce this?

**Comment by OliverJAsh** [comment]
It would also be good if the auto fix for the `"arrow-return-shorthand": [true, "multiline"]` rule respected this setting.

**Comment by JoshuaKGoldberg** [comment]
> add a rule to enforce this?

I'm -0 on including this kind of Prettier-style formatting in TSLint core. Is there a lot of need for such a thing?

> It would also be good if the auto fix for the "arrow-return-shorthand": [true, "multiline"] rule respected this setting.

Rules intentionally don't have visibility onto each other's configuration settings to help keep them simpler. #2556 discusses better overlapping fix handling; in the meantime you can run `--fix` multiple times to resolve everything.

**Comment by JoshuaKGoldberg** [comment]
Per #4534, closing this formatting rule request. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!
